### DISCUSSION TITLE
Current Windows 11 stable release is 22000.438, not 22000.434, and info about Windows 11 version updates

### DISCUSSION THREAD

**Discussion by yetisyny** [valinet/ExplorerPatcher#716] [General]
I’m on the latest stable release of Windows 11, version 22000.438. The program works fine. However, the current version number for ExplorerPatcher has “434” in it instead of “438”. You can see the list of Windows 11 builds and when they were released at https://docs.microsoft.com/en-us/windows/release-health/windows11-release-information.

Anyway, hardly anything was changed between Windows 11 versions 434 and 438, just a very short list of things at https://support.microsoft.com/en-us/topic/january-17-2022-kb5010795-os-build-22000-438-out-of-band-2d2b9310-d845-41c4-9907-aeea24f36a63 and nothing there seems to affect ExplorerPatcher at all. So I would suggest updating the version number of ExplorerPatcher in the future pre-release builds from having “434” in it to “438” because I can confirm it doesn’t have any problems on build 438, at least not any that wouldn’t also be present in build 434 too.

If you look back at the earlier builds, most of them don’t make too many changes, but Windows 11 builds 282 and 348 both stand out as having very long lists of many changes they made, the only 2 that are like that... both of them are labelled “Preview”, while the releases that do less changes either do not have labels or are labelled “Out-of-band”. Apparently the “Preview” releases of Windows 11 are also called C releases and they make the most changes to functionality that programs like this would need to worry about. There was not a C release in December but they are supposed to resume January (the current month), I would expect one any day now. The release in December, build 376, which was not labelled, is considered a B release, a monthly security release which is mostly security fixes but those also seem to typically fix one or two known issues in Windows 11 (but not a long list like the C releases). The “Out-of-band” release that recently happened is, I am guessing, a release that is pushed out, not following any pre-set schedule like the B and C releases, to fix known issues, and I think it is probably the first A release. So the C releases are sort of like service packs or cumulative update packages, the B releases are sort of like monthly security updates, and the A releases are sort of like small updates that are pushed out with higher priority not following a schedule to fix important issues.

Anyway I thought this might help in telling which of the Windows 11 updates in the future are the ones that you need to look out for breaking compatibility the most and which are least likely to break compatibility. The ones that are labelled “Preview”, the C updates like build 348, whose page explains this, are the ones to look out for, the ones that are most likely to break things because they make large numbers of changes that are not security fixes: https://support.microsoft.com/en-us/topic/november-22-2021-kb5007262-os-build-22000-348-preview-7f3e18d7-4189-4882-b0e9-afc920253aee. It seems there will be one of them a month, most months, but they skipped it in December and the one for this current January hasn’t happened yet. The 2 that did happen were October 21 and November 22 in 2021 so since today is January 23, I would expect one soon based on the pattern of those dates.

**Comment by valinet** [comment]
Thank you for the explanation, to be honest, I never really understood their nomenclature. So, basically:

* `A` means random releases when they feel something has to be fixed right away.
* `B` are regular Patch Tuesday updates
* `C` are preview releases of the upcoming `B` release

I tested it on 22000.348 in a virtual machine as well, and similarly saw that it worked just fine there. I haven't changed the version number through because when I check for updates on my working PC (for which I just have it install updates on-demand, only stable ones, no Insider etc), I was offered only up to build 22000.343. Indeed, "KB5010795" is listed, but as an opt-in download:

![image](https://user-images.githubusercontent.com/6503598/150678588-d8dcfaf9-093e-42f5-96d1-aa9fbbe829e0.png)

So I supposed most user wouldn't be installing that, and also that the build was another preview. Or is it just that they push updates only on Patch Tuesday now, and all the rest are just suggested throughout the month, but not offered generally (automatically installed)?

I also made a clean Windows 11 install in a VM, with the ISO you can download from Microsoft, to see what version it would actually install after performing all its updates. It indeed offered 434 as well, with 438 suggested in a similar note as the one above.

So yeah, this is where it all comes from, I just thought they do Patch Tuesday and preview of Patch Tuesday releases, and decided to stick version numbers to Patch Tuesday generally. Now, it may be a good idea to bump the version up once we can kind of confirm it works on whatever stable newer version we find. I also checked on 22000.466 for example, and there it works just fine as well. So, in the end, I bumped up the version all the way to 22000.466.

Thanks for the info. Idk, hopefully I will make a stable release soon. I kind of have a rule of it being okay to mark a pre-release as a release after about 2 days of making no commits to the repo, this being like an indicator that what we have there is relatively stable.
